kubectl-volsync/cmd/migration_create.go

```diff
@@ -72,34 +108,294 @@ func validateMigrationCreate(cmd *cobra.Command, args []string) error {
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      mrd.PVCName,
+				Namespace: mrd.Namespace,
+			},
+			Spec: v1.PersistentVolumeClaimSpec{
+				AccessModes:      mrd.Destination.AccessModes,
+				StorageClassName: mrd.Destination.StorageClassName,
+				Resources: v1.ResourceRequirements{
+					Requests: v1.ResourceList{
+						v1.ResourceStorage: *mrd.Destination.Capacity,
+					},
+				},
+			},
+		}
+	}
+	if err := mc.clientObject.Create(ctx, destPVC); err != nil {
+		return nil, err
+	}
+	mrd.Destination.StorageClassName = destPVC.Spec.StorageClassName
+	klog.Infof("Created destination PVC: \"%s\"", mrd.PVCName)
+
+	return destPVC, nil
+}
+
+func (mc *migrationCreate) getDestinationPVC(ctx context.Context) *v1.PersistentVolumeClaim {
+	mrd := mc.mr.data.Destination
+	destPVC := &v1.PersistentVolumeClaim{}
+	pvcInfo := types.NamespacedName{
+		Namespace: mrd.Namespace,
+		Name:      mrd.PVCName,
+	}
+	err := mc.clientObject.Get(ctx, pvcInfo, destPVC)
+	if err == nil {
+		return destPVC
+	}
+
+	return nil
+}
+
+func (mc *migrationCreate) createDestination(ctx context.Context) error {
+	mrd := mc.mr.data.Destination
+	rsyncSpec := &volsyncv1alpha1.ReplicationDestinationRsyncSpec{
+		ReplicationDestinationVolumeOptions: volsyncv1alpha1.ReplicationDestinationVolumeOptions{
+			CopyMethod:     mrd.Destination.CopyMethod,
+			DestinationPVC: &mrd.PVCName,
+		},
+		ServiceType: mrd.Destination.ServiceType,
+	}
+	rd := &volsyncv1alpha1.ReplicationDestination{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      mrd.MDName,
+			Namespace: mrd.Namespace,
+		},
+		Spec: volsyncv1alpha1.ReplicationDestinationSpec{
+			Rsync: rsyncSpec,
+		},
+	}
+	if err := mc.clientObject.Create(ctx, rd); err != nil {
 		return err
 	}
-	relation.Set("capacity", cap)
 
-	if err = relation.Save(); err != nil {
-		return fmt.Errorf("unable to save relationship configuration: %w", err)
+	// wait for migrationdestination to become ready
+	nsName := types.NamespacedName{
+		Namespace: mrd.Namespace,
+		Name:      mrd.MDName,
 	}
+	rd = &volsyncv1alpha1.ReplicationDestination{}
+	err := wait.PollImmediate(5*time.Second, 2*time.Minute, func() (bool, error) {
+		err := mc.clientObject.Get(ctx, nsName, rd)
+		if err != nil {
+			return false, err
+		}
+		if rd.Status == nil {
+			return false, nil
+		}
+		if rd.Status.Rsync.Address == nil {
+			klog.Infof("Waiting for MigrationDestination %s RSync address to populate", rd.Name)
+			return false, nil
+		}
 
-	// TODO: Set up objects on the cluster
-	_, _ = newClient("")
+		if rd.Status.Rsync.SSHKeys == nil {
+			klog.Infof("Waiting for MigrationDestination %s RSync sshkeys to populate", rd.Name)
+			return false, nil
+		}
+
+		klog.Infof("Found MigrationDestination RSync Address: %s", *rd.Status.Rsync.Address)
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+	mrd.Destination.Address = rd.Status.Rsync.Address
+	mrd.Destination.Port = rd.Status.Rsync.Port
+	mrd.Destination.SSHKeys = rd.Status.Rsync.SSHKeys
```

Comment:
I would suggest fetching these from the cluster when we want to use them as opposed to saving them in the relationship file. Though waiting for them to populate (as above) is probably good.

Reply:
Okay, sure. I will remove this if not required to be saved as part of relationship file